### PR TITLE
fix: Specify row count in sort_batch for batch with no columns

### LINKED ISSUE #10093: Got arrow error when sorting on empty batch
### Describe the bug

Found the issue when Comet tries to run a Spark query which sorts empty batch: https://github.com/apache/arrow-datafusion-comet/issues/270

```
Arrow error: Invalid argument error: must either specify a row count or at least one column
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #10093.

### CODE REVIEW

File: datafusion/physical-plan/src/sorts/partial_sort.rs [hunk 1]
```diff
@@ -390,6 +390,7 @@ impl PartialSortStream {
     fn sort_in_mem_batches(self: &mut Pin<&mut Self>) -> Result<RecordBatch> {
         let input_batch = concat_batches(&self.schema(), &self.in_mem_batches)?;
         self.in_mem_batches.clear();
+        println!("input_batch: {:?}", input_batch);
```
Comment (andygrove) [inline datafusion/physical-plan/src/sorts/partial_sort.rs:393]: Looks like some debug logging that needs to be removed?

Comment (viirya) [reply datafusion/physical-plan/src/sorts/partial_sort.rs:393]: Oops, yea.

**Review by alamb — APPROVED** [review]
Seems good to me -- thanks @viirya 

If the usecase is sorting a batch with 0 columns, I do wonder if that could be optimized away somehow.

**Comment by viirya** [comment]
> Seems good to me -- thanks @viirya
> 
> If the usecase is sorting a batch with 0 columns, I do wonder if that could be optimized away somehow.

Yea, that's good idea. The physical query plan goes through Spark optimizer. It seems there is no corresponding optimization rule.

**Pull request merged**